src/metabase/pulse/util.clj

```diff
@@ -14,6 +14,7 @@
    [toucan2.core :as t2]))
 
 ;; TODO - this should be done async
+;; TODO - this and `execute-multi-card` should be made more efficient: eg. we query for the card several times
```

Comment:
I think that in order to make these execute fns more efficient we should change the contract to require passing in the instance, so that we fetch earlier and only once.

Not needed for this PR, but an idea anyway.